roles/packer-setup/lookup_plugins/windows_update.py

```diff
@@ -259,13 +259,19 @@ def get_download_urls(self):
             headers = {
                 'Content-Type': 'application/x-www-form-urlencoded',
             }
-            with urlopen('%s/DownloadDialog.aspx' % CATALOG_URL, data=data,
-                         headers=headers) as resp:
-                resp_text = to_text(resp.read()).strip()
-
-            link_matches = re.findall(DOWNLOAD_PATTERN, resp_text)
-            if len(link_matches) == 0:
-                raise ValueError("Failed to find any download links for '%s'" % str(self))
+            linkFound = False
+            while not linkFound:
+                with urlopen('%s/DownloadDialog.aspx' % CATALOG_URL, data=data,
+                            headers=headers, timeout=120) as resp:
+                    resp_text = to_text(resp.read()).strip()
+
+                link_matches = re.findall(DOWNLOAD_PATTERN, resp_text)
+                if len(link_matches) == 0:
+                    # raise ValueError("Failed to find any download links for '%s'" % str(self))
+                    #display.v("Download link not found - read it again")
+                    linkFound = False
```

Comment:
This doesn't sound right, I thought this issue was a timeout occurring on trying to open the URL. This implies that it is returning different data on a request.

Also we run the risk of the code never exiting this loop if the match never occurs causing a hang without any indication of what has happened.

Reply:
Yep, right. From my side, this was just a quick patch to make it working (I´ve spent some time searching for the core of the error already). Making it part of the main trunk would be nicer to add some verifications as you said. On the other hand - this part is fully dependent on the structure of the page returned and there is always a chance some change will make it non-functional. So, maybe, only limiting the number of attempts would be enough...